### PR TITLE
adding build for fmt

### PR DESCRIPTION
**Pull request by cmhamel**

### CODE REVIEW

File: F/Fmt/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+script = raw"""
+INSTALL_PATH=${prefix}
+cd $WORKSPACE/srcdir/fmt
+mkdir build
+cd build
+cmake \
+-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+-DBUILD_SHARED_LIBS=True \
+-DCMAKE_INSTALL_PREFIX=${INSTALL_PATH} \
+..
```
Comment (giordano) [inline F/Fmt/build_tarballs.jl:23]: Why defining a variable, used only once, equal to another variable?  Why not directly using that variable, like in all other recipes here?  Also, you keep not following the [documentation](https://docs.binarybuilder.org/stable/build_tips/#CMake-builds), use also `-DCMAKE_BUILD_TYPE=Release`: 
```suggestion
cd $WORKSPACE/srcdir/fmt
mkdir build
cd build
cmake \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    -DBUILD_SHARED_LIBS=True \
    -DCMAKE_INSTALL_PREFIX=${prefix} \
    ..
```

File: F/Fmt/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+make
+make install
+
+# the license file isn't found automatically for some reason
+# perhaps it's the .rst extension?
+#
+mkdir ${prefix}/share
+mkdir ${prefix}/share/licenses
+mkdir ${prefix}/share/licenses/Fmt
+cp $WORKSPACE/srcdir/fmt/LICENSE.rst ${prefix}/share/licenses/Fmt/LICENSE
```
Comment (giordano) [inline F/Fmt/build_tarballs.jl:33]: You might really want to [read the documentation](https://docs.binarybuilder.org/stable/build_tips/#Installing-the-license-file)
```suggestion
install_license ../LICENSE.rst
```

File: F/Fmt/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+platforms = expand_cxxstring_abis(platforms)
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libfmt", :libfmt),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
```
Comment (giordano) [inline F/Fmt/build_tarballs.jl:53]:
```suggestion
dependencies = Dependency[
```

File: F/Fmt/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
+
```
Comment (giordano) [inline F/Fmt/build_tarballs.jl:58]:
```suggestion
```

File: F/Fmt/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+# Bash recipe for building across all platforms
+script = raw"""
+
+cd $WORKSPACE/srcdir/fmt
+mkdir build
+cd build
+
+cmake \
+-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_gcc.cmake \
```
Comment (giordano) [inline F/Fmt/build_tarballs.jl:21]: Why using GCC everywhere?

Comment (cmhamel) [reply F/Fmt/build_tarballs.jl:21]: I run into clang errors on mac without this. Linux and windows build fine either way but Mac is the problem child in this case.

Comment (giordano) [reply F/Fmt/build_tarballs.jl:21]: Can you elaborate on _what_ errors you got?

Comment (cmhamel) [reply F/Fmt/build_tarballs.jl:21]: ld: library not found for -lc++fs
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)

Above is the error encountered on mac (both x86_64 and aarch64).

File: F/Fmt/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+-DCMAKE_BUILD_TYPE=Release \
+-DBUILD_SHARED_LIBS=True \
+-DCMAKE_INSTALL_PREFIX=${prefix} \
+..
```
Comment (giordano) [inline F/Fmt/build_tarballs.jl:25]: Don't build the tests
```suggestion
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    -DBUILD_SHARED_LIBS=True \
    -DFMT_TEST=OFF \
    -DCMAKE_INSTALL_PREFIX=${prefix} \
    ..
```

Also, I'm moderately sure I [already suggested](https://github.com/JuliaPackaging/Yggdrasil/pull/5565#discussion_r979091238) using some indentation before.

**Pull request merged**